ddtrace/tracer/tracer.go

```diff
@@ -451,7 +465,13 @@ func (t *tracer) pushChunk(trace *chunk) {
 	select {
 	case t.out <- trace:
 	default:
-		log.Error("payload queue full, dropping %d traces", len(trace.spans))
+		log.Debug("payload queue full, trace %d dropped %d spans", trace.traceID, len(trace.spans))
+		if !trace.dropped {
+			t.totalTracesDropped.mu.Lock()
+			defer t.totalTracesDropped.mu.Unlock()
```

Comment:
In this case, the `defer` is fine because if we are at line 470, we only have a few lines of code until this entire function scope completes, but generally speaking, it's best to "release" a lock as soon as possible (that is, after reading/writing to the value) to minimize contention